pyquil/compatibility/v2/api/_quantum_computer.py

```diff
@@ -0,0 +1,776 @@
+##############################################################################
+# Copyright 2021 Rigetti Computing
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+##############################################################################
+import itertools
+import warnings
+from math import log, pi
+from typing import Any, Dict, List, Mapping, Optional, Sequence, Set, Tuple, Union, cast
+
+import numpy as np
+from pyquil.api._compiler import AbstractCompiler
+from pyquil.api._qam import QAM
+from pyquil.api._qpu import QPU
+from pyquil.api._quantum_computer import QuantumComputer as QuantumComputerV3
+from pyquil.api._quantum_computer import get_qc as get_qc_v3
+from pyquil.api._qvm import QVM
+from pyquil.experiment._main import Experiment
+from pyquil.experiment._memory import merge_memory_map_lists
+from pyquil.experiment._result import ExperimentResult, bitstrings_to_expectations
+from pyquil.experiment._setting import ExperimentSetting
+from pyquil.gates import MEASURE, RX
+from pyquil.paulis import PauliTerm
+from pyquil.quil import Program, validate_supported_quil
+from pyquil.quilatom import qubit_index
+from qcs_api_client.client import QCSClientConfiguration
+from rpcq.messages import PyQuilExecutableResponse, QuiltBinaryExecutableResponse
+
+from ._qam import StatefulQAM
+
+ExecutableDesignator = Union[QuiltBinaryExecutableResponse, PyQuilExecutableResponse]
+
+
+class QuantumComputer(QuantumComputerV3):
+    compiler: AbstractCompiler
+    qam: StatefulQAM
+
+    def __init__(
+        self,
+        *,
+        name: str,
+        qam: QAM,
+        device: Any = None,
+        compiler: AbstractCompiler,
+        symmetrize_readout: bool = False,
+    ) -> None:
+        """
+        An interface designed to ease migration from pyQuil v2 to v3, and compatible with most
+        use cases for the pyQuil v2 QuantumComputer.
+
+        A quantum computer for running quantum programs.
+
+        A quantum computer has various characteristics like supported gates, qubits, qubit
+        topologies, gate fidelities, and more. A quantum computer also has the ability to
+        run quantum programs.
+
+        A quantum computer can be a real Rigetti QPU that uses superconducting transmon
+        qubits to run quantum programs, or it can be an emulator like the Rigetti QVM with
```

Comment:
nit: Technically the QVM is no longer "Rigetti's" right?

Reply:
nice catch